src/xcode/ENA/ENA/Source/Services/OTP/OTPService.swift

```diff
@@ -0,0 +1,113 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+protocol OTPServiceProviding {
+	/// Checks if there is a valid stored otp. If so, we check if we can reuse it beacuse it was not already used, or if it was already used. If so, we return a failure.  If there is not a stored otp token, or if the stored token's expiration date is reached, a new fresh otp token is generated and stored.
+	/// After these validation checks, the service tries to authorize the otp against the server.
+	/// - Parameters:
+	///   - ppacToken: a generated and valid PPACToken from the PPACService
+	///   - completion: The completion handler
+	/// - Returns:
+	///   - success: the authorized and stored otp as String
+	///   - failure: an OTPError, for which the caller can build a dedicated error handling
+	func getValidOTP(ppacToken: PPACToken, completion: @escaping (Result<String, OTPError>) -> Void)
+	/// discards any stored otp.
+	func discardOTP()
+}
+
+final class OTPService: OTPServiceProviding {
+
+	// MARK: - Init
+
+	init(
+		store: Store,
+		client: Client
+	) {
+		self.store = store
+		self.client = client
+	}
+
+	// MARK: - Protocol OTPServiceProviding
+
+	func getValidOTP(ppacToken: PPACToken, completion: @escaping (Result<String, OTPError>) -> Void) {
```

Comment:
Since there is no API to get an invalid token I would propose to rename it to `getOTP(...)` or `otp(for  ppacToken: ...)`

Reply:
good point. I will change that.